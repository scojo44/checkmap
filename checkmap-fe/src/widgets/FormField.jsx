import React from 'react'

export default function FormField({name, label, inputType = "text", register, validation, errors}) {
  const type = inputType != "text"? {type: inputType} : {}; // Leaves out type="text"

  return (
    <p>
      <label htmlFor={name}>{label}: </label>
      <input {...type} {...register(name, validation)} />
      {errors[name] && <span className="input-error"> {errors[name].message}</span>}
    </p>
  );
}
