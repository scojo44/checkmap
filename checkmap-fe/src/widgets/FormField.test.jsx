import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import FormField from './FormField'

describe('FormField Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <FormField name="test" label="Test Label" register={x => x} errors={{test: {message: 'test message'}}}/>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <FormField name="test" label="Test Label" register={x => x} errors={{test: {message: 'test message'}}}/>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
