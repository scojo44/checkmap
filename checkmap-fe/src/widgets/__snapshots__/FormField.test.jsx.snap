// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormField Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="FormField"
  >
    <label
      for="test"
    >
      Test Label: 
    </label>
    <input />
    <span
      class="input-error"
    >
       test message
    </span>
  </p>
</DocumentFragment>
`;
